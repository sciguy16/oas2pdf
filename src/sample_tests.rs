use crate::{args::Args, do_run, test::rand_str};
use std::path::Path;

fn run_sample_test<P: AsRef<Path>>(input: P) -> String {
    let typst_path = format!("/tmp/sample-typst-out-{}", rand_str());
    let mut args = Args {
        out: Some((&typst_path).into()),
        typst: true,
        input: input.as_ref().into(),
        save_template: None,
        template: None,
    };
    do_run(&args).unwrap();
    // Store the typst file for snapshot testing
    let rendered = std::fs::read_to_string(&typst_path).unwrap();

    // try rendering the PDF to check for syntax errors
    args.typst = false;
    do_run(&args).unwrap();

    std::fs::remove_file(typst_path).unwrap();
    rendered
}

#[test]
fn docker_api() {
    insta::assert_snapshot!(run_sample_test("samples/docker.yaml"));
}
