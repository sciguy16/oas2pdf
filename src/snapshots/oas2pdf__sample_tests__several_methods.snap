---
source: src/sample_tests.rs
expression: "run_sample_test(\"samples/several-methods.yaml\")"
---
#let title = [ Several Methods API ]

#set heading(numbering: "1.")
#set page(
  header: align(
    right + horizon,
    title
  ),
  numbering: "1/1"
)

#align(horizon)[
	#align(center, text(27pt)[
	  *#title*
	])
	#align(center, text(24pt)[ A simple API with different verbs ])
]

#pagebreak()

#outline(
	depth: 2,
	indent: auto,
)


#pagebreak()
= Container

== /get-post

=== get - GetThings
Get things

Get things

==== Parameters
#table(
	columns: 4,
	[Name], [Required], [Description], [Schema],

	[all],
	[false],
	[Show all containers. Only running containers are shown by default (i.e., this defaults to false)],
	[],

)
